.github/workflows/io-test.yml

```diff
@@ -33,49 +33,51 @@ jobs:
+          echo "::set-output name=HASH_EPHY_DATASET::$(git ls-remote https://gin.g-node.org/NeuralEnsemble/ephy_testing_data.git HEAD | cut -f1)"
+
+      - uses: actions/cache@v3
+        # Loading cache of ephys_testing_dataset
+        id: cache-datasets
+        with:
+          path: ~/ephy_testing_data
+          key: ${{ runner.os }}-datasets-${{ steps.vars.outputs.HASH_EPHY_DATASET }}
+
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: neo-test-env
+          python-version: ${{ matrix.python-version }}
+          clean-patched-environment-file: false
+
+      - uses: actions/cache@v3
         # the cache for python package is reset:
         #   * every month
         #   * when requirements/requirements_testing change
-        id: cache-venv
+        id: cache-conda-env
         with:
-          path: ~/test_env
-          key: ${{ runner.os }}-venv-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/requirements_testing.txt') }}-${{ steps.date.outputs.date }}
-          restore-keys: |
-            ${{ runner.os }}-venv-
+          path: /usr/share/miniconda/envs/neo-test-env
+          key: ${{ runner.os }}-conda-env-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/requirements_testing.txt') }}-${{ hashFiles('**/environment_testing.txt') }}-${{ steps.date.outputs.date }}
+
+      - name: Install testing dependencies
+        # testing environment is only installed if no cache was found
+        if: steps.cache-conda-env.outputs.cache-hit != 'true'
+        run: |
+          conda env update neo-test-env --file environment_testing.yml
 
-      - name: Install dependencies
+      - name: Configure git
         run: |
-          # this is for datalad and download testing datasets
-          sudo apt install git git-annex
-          # needed for correct operation of git/git-annex/DataLad
           git config --global user.email "neo_ci@fake_mail.com"
           git config --global user.name "neo CI"
-          # create an environment and install everything
-          python -m venv ~/test_env
-          source ~/test_env/bin/activate
-          python -m pip install --upgrade pip
-          # pip install -r requirements.txt
-          pip install -r requirements_testing.txt
-          pip install -e .
 
-      - name: Get ephy_testing_data current head hash
-        # the key depend on the last comit repo https://gin.g-node.org/NeuralEnsemble/ephy_testing_data.git
-        id: vars
+      - name: Install neo
         run: |
-          echo "::set-output name=HASH_EPHY_DATASET::$(git ls-remote https://gin.g-node.org/NeuralEnsemble/ephy_testing_data.git HEAD | cut -f1)"
-
-      - uses: actions/cache@v2
-        id: cache-datasets
-        with:
-          path: ~/ephy_testing_data
-          key: ${{ runner.os }}-datasets-${{ steps.vars.outputs.HASH_EPHY_DATASET }}
-          restore-keys: |
-            ${{ runner.os }}-datasets
+          pip install --upgrade -e .
```

Comment:
Are you sure that this upgrade option do recurssively also upgrade numpy and quantities ?
The upgrade rules have change a bit.

Reply:
Even if it does not do recursive upgrades with the next month the default versions of these packages will be updated. Do you think this lag is too long? If you want to force recursive dependency upgrades I can add a `--upgrade-strategy "eager"`, but this means the tests might have a longer runtime until a new month starts.

Reply:
ok